### PR TITLE
Feature/58 camera panning

### PR DESCRIPTION
**Pull request by JuiceGrape**
## Summary
Added camera panning and jumping. It works in pixels instead of unity units to make it easier for writers. It automatically continues, so writers need to put a WAIT action after to make sure it waits.

## Before/after screenshots and/or animated gif
scrollybois

## Testing instructions
Play the inky test scene

## Additional information

- `[ ]` Changes UI
- `[x]` Introduces new feature
- `[ ]` Removes existing feature
- `[ ]` Has associated resource:

### CODE REVIEW

**Review by PhoebeMitchell — DISMISSED** [review]
Looks good. The "actor not assigned" errors kind of bug me. Is that something that will be fixed when HIDE_ACTOR is implemented? If not I think they should be warnings as errors imply something that needs to be fixed for me. Also the animators on the actors in the new BG scenes could be disabled to stop the no animator controller warning.

**Comment by JuiceGrape** [comment]
Discussed with moths: No active actor being present should be a legal state for the system.

File: Assets/Scripts/Actor/ActorController.cs [hunk 1]
```diff
@@ -61,7 +62,8 @@ public void SetActiveActor(string actor)
     {
         try
         { 
-            _activeActor.SetActor(_actorInventory[actor]);
+            if (_activeActor != null)
```
Comment (JuiceGrape) [inline Assets/Scripts/Actor/ActorController.cs:65]: Discussed with moths: No active actor being present should be a legal state for the system.

Comment (JuiceGrape) [reply Assets/Scripts/Actor/ActorController.cs:65]: This convo is here to prevent merging

**Review by PhoebeMitchell — DISMISSED** [review]

File: Assets/Scripts/Actor/ActorController.cs (hunk 1 above)
Comment (JuiceGrape) [reply Assets/Scripts/Actor/ActorController.cs:65]: Done

File: Assets/Scripts/BGScenes/SceneController.cs [hunk 2]
```diff
@@ -51,9 +55,32 @@ public void FadeOut(float seconds)
...
+
+    /// <summary>
+    /// Pans the camera position to the target position
+    /// </summary>
+    /// <param name="pos">Target position</param>
+    /// <param name="time">Time for the pan to take in seconds</param>
+    /// <returns>IEnumerator stuff for coroutine</returns>
+    private IEnumerator PanToPosition(Vector2 targetPos, float time)
+    {
+        var startPos = _activeScene.transform.position;
```
Comment (lyndsiWilliams) [inline Assets/Scripts/BGScenes/SceneController.cs]: Can these `var`s be defined, or do they need to stay flexible?

File: Assets/Scripts/BGScenes/SceneController.cs [hunk 3]
```diff
@@ -51,9 +55,32 @@ public void FadeOut(float seconds)
...
+        var t = 0f;
```
Comment (lyndsiWilliams) [inline Assets/Scripts/BGScenes/SceneController.cs]: Can you use a more descriptive variable name than `t` here?

File: Assets/Scripts/BGScenes/SceneController.cs [hunk 4]
```diff
@@ -130,4 +161,9 @@ public void CancelWaitCoroutine()
         StopCoroutine(_waitCoroutine);
         _waitCoroutine = null;
     }
+
+    public Vector2 PixelPositionToUnitPosition(Vector2Int pixelPosition)
```
Comment (lyndsiWilliams) [inline Assets/Scripts/BGScenes/SceneController.cs:206]: This method is missing a summary ✨

**Review by lyndsiWilliams — COMMENTED** [review]
Looks great! Just a couple of nits ✨

**Review by lyndsiWilliams — DISMISSED** [review]
Oops, meant to do the above comments as an "approve" with comments.

File: Assets/Scripts/BGScenes/SceneController.cs (hunk 2 above)
Comment (JuiceGrape) [reply Assets/Scripts/BGScenes/SceneController.cs]: Cleaned it up

**Review by PhoebeMitchell — APPROVED** [review]

**Review by lyndsiWilliams — APPROVED** [review]
Beautiful!

**Pull request merged**